cli/src/main/java/com/devonfw/tools/ide/tool/plugin/ToolPlugins.java

```diff
@@ -0,0 +1,76 @@
+package com.devonfw.tools.ide.tool.plugin;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.devonfw.tools.ide.context.IdeContext;
+
+/**
+ * A wrapper class for holding two maps of plugin descriptors: one keyed by plugin ID and the other keyed by plugin name.
+ */
+public class ToolPlugins {
+
+  private final Map<String, ToolPluginDescriptor> mapById;
+  private final Map<String, ToolPluginDescriptor> mapByName;
+
+
+  private final IdeContext context;
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public ToolPlugins(IdeContext context) {
+    super();
+    this.context = context;
+    this.mapById = new HashMap<>();
+    this.mapByName = new HashMap<>();
+  }
+
+  /**
+   * @return the {@link ToolPluginDescriptor} for the given {@link ToolPluginDescriptor#id() ID} or {@code null} if not found.
+   */
+  public ToolPluginDescriptor getById(String id) {
+    return this.mapById.get(id);
+  }
+
+  /**
+   * @return the {@link ToolPluginDescriptor} for the given {@link ToolPluginDescriptor#name() name} or {@code null} if not found.
+   */
+  public ToolPluginDescriptor getByName(String name) {
+    return this.mapByName.get(name);
+  }
+
+  /**
+   * @return the immutable {@link Collection} of {@link ToolPluginDescriptor}s configured for this IDE tool.
+   */
+  public Collection<ToolPluginDescriptor> getPlugins() {
+
+    Map<String, ToolPluginDescriptor> map = this.mapById;
+    if (map.isEmpty()) {
+      map = this.mapByName; // potentially plugins for this tool have no ID
+    }
+    return Collections.unmodifiableCollection(map.values());
+  }
+
+  /**
+   * Registers a {@link ToolPluginDescriptor} to this map.
+   */
+  protected void add(ToolPluginDescriptor descriptor) {
+    put(descriptor.name(), descriptor, this.mapByName);
+    String id = descriptor.id();
+    if ((id != null) && !id.isEmpty()) {
+      put(id, descriptor, this.mapById);
+    }
+  }
+
+  public void put(String key, ToolPluginDescriptor descriptor, Map<String, ToolPluginDescriptor> map) {
```

Comment:
Thanks for pointing me here. This is not a public/API method. I simply made it private since it only exists to make the `add` method more compact and avoid redundancies.